util/iosutil/iosutil.go

```diff
@@ -0,0 +1,78 @@
+package iosutil
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+)
+
+// Version specifies the version of an iOS device.
+type Version struct {
+	Major int
+	Minor int
+}
+
+// ParseVersion parses the major.minor version for an iOS device.
+func ParseVersion(v string) (Version, error) {
+	parts := strings.Split(v, ".")
+
+	if len(parts) != 2 {
+		return Version{}, errors.New("expected major.minor format")
+	}
+
+	major, err := strconv.Atoi(parts[0])
+	if err != nil {
+		return Version{}, errors.New("major version is not an integer")
+	}
+
+	minor, err := strconv.Atoi(parts[1])
+	if err != nil {
+		return Version{}, errors.New("minor version is not an integer")
+	}
+
+	version := Version{
+		Major: major,
+		Minor: minor,
+	}
+	return version, nil
+}
+
+// EqualOrGreater returns true if the iOS device version is equal or greater to the desired version, using semantic versioning.
+func (v Version) EqualOrGreater(major, minor int) bool {
+	if v.Major == major {
+		return v.Minor >= minor
+	}
+
+	return v.Major > major
+}
+
+// VersionClassification describes iOS version classifications which are important to Prebid Server.
+type VersionClassification int
+
+// Values of VersionClassification.
+const (
+	VersionUnknown VersionClassification = iota
+	Version140
+	Version141
+	Version142OrGreater
+)
+
+// DetectVersionClassification detects the iOS version classification.
+func DetectVersionClassification(v string) VersionClassification {
+	// exact comparisons first. no parsing required.
+	if v == "14.0" {
+		return Version140
+	}
+	if v == "14.1" {
```

Comment:
Are we positive the `req.Device.OSV` will come with iOS versions written exactly like `"14.0"` and `"14.1"`? Do you think it'd be safer to try to match with a regexp instead? Or maybe use `string` library functions such as `string.Contains()` or `string.HasPrefix()`

Reply:
Yes. I asked in the linked issue if that was a valid assumption and the author confirmed it was, including a link back to the iOS version docs.